src/Sylius/Component/Channel/spec/Context/SingleChannelContextSpec.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Channel\Context\SingleChannel;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Context\ChannelNotFoundException;
+use Sylius\Component\Channel\Context\SingleChannelContext;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+
+/**
+ * @mixin SingleChannelContext
+ *
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class SingleChannelContextSpec extends ObjectBehavior
+{
+    function let(ChannelRepositoryInterface $channelRepository)
+    {
+        $this->beConstructedWith($channelRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Channel\Context\SingleChannel\ChannelContext');
+    }
+
+    function it_implements_channel_context_interface()
+    {
+        $this->shouldImplement(ChannelContextInterface::class);
+    }
+
+    function it_returns_a_channel_if_it_is_the_only_one_defined(
+        ChannelRepositoryInterface $channelRepository,
+        ChannelInterface $channel
+    ) {
+        $channelRepository->findAll()->willReturn([$channel]);
+
+        $this->getChannel()->shouldReturn($channel);
+    }
+
+    function it_throws_an_channel_not_found_exception_if_there_are_no_channels_defined(
```

Comment:
I'm totally agree for above case in [CompositeChannelContextSpec](https://github.com/Sylius/Sylius/pull/4056/files#diff-cd860143795fd6ba7770e844de50c2ceR37). But here we have just different values returned by repository function. Won't it be better?

```php
    function it_throws_an_channel_not_found_exception_if_there_isnt_one_channel_defined(
        ChannelRepositoryInterface $channelRepository
    ) {
        $channelRepository->findAll()->willReturn([], [$firstChannel, $secondChannel]);

        $this->shouldThrow(ChannelNotFoundException::class)->during('getChannel');
        $this->shouldThrow(ChannelNotFoundException::class)->during('getChannel');
    }
```